autoprecompiles/src/pgo/cell/mod.rs

```diff
@@ -1,36 +1,47 @@
 use std::{
+    cmp::Ordering,
     collections::{BTreeMap, HashMap},
     io::BufWriter,
     path::Path,
     sync::{Arc, Mutex},
 };
 
-use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use priority_queue::PriorityQueue;
+use rayon::iter::{ParallelBridge, ParallelIterator};
 use serde::{Deserialize, Serialize};
 
 use crate::{
     adapter::{Adapter, AdapterApc, AdapterApcWithStats, AdapterVmConfig, PgoAdapter},
-    blocks::BasicBlock,
+    blocks::Block,
     evaluation::EvaluationResult,
-    pgo::cell::selection::parallel_fractional_knapsack,
+    execution_profile::ExecutionProfile,
     PowdrConfig,
 };
 
-mod selection;
-
-pub use selection::KnapsackItem;
```

Comment:
I'm wondering if we could keep the selection in a different trait still so it's easy to test in the same way we currently test the knapsack selection. Superblocks seems like a generalisation of knapsack in which if you take some object, then the value of taking some other objects diminishes. Like if you already have a swiss knife, a knife is less valuable.